syntax = "proto3";

package test3;

enum OuterEnum {
  UNKNOWN = 0;
  FOO3 = 1;
  BAR3 = 2;
}

message OuterMessage3 {
  bool a_bool = 1;
}

message SimpleProto3 {
  string a_string = 1;
  repeated string a_repeated_string = 2;
  OuterEnum a_outer_enum = 3;
  OuterMessage3 outer_message = 4;

  oneof a_oneof {
      string a_oneof_1 = 5;
      string a_oneof_2 = 6;
  }

  oneof b_oneof {
      string b_oneof_1 = 7;
      string b_oneof_2 = 8;
  }

  OuterMessage3 bool = 9;

  // Test having fieldname match messagename
  OuterEnum OuterEnum = 10;
  OuterMessage3 OuterMessage3 = 11;
}
